styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -547,19 +550,30 @@ private void setupMetrics(
                   .count()));
       stats.registerActiveStates(
           state,
-          "none",
-          () -> stateManager.activeStates().values().stream()
+          "none", () -> stateManager.activeStates().values().stream()
               .filter(runState -> runState.state().equals(state))
               .filter(runState -> !runState.data().trigger().isPresent())
               .count());
     });
 
-    stats.registerQueuedEvents(queuedEventsCount);
-    stats.registerWorkflowCount("all", allWorkflowsCount);
-    stats.registerWorkflowCount("configured", configuredWorkflowsCount);
-    stats.registerWorkflowCount("enabled", configuredEnabledWorkflowsCount);
-    stats.registerWorkflowCount("docker_termination_logging_enabled",
-                                dockerTerminationLoggingEnabledWorkflowsCount);
+    final Function<String, Gauge<Long>> createResourceConfiguredGauge = (resource) -> {
+      final Supplier<Optional<Resource>> resourceSupplier =
```

Comment:
Why `Optional<Resource>`? A `CachedSupplier` pattern that prevents all `orElse` should be possible if the resource isn't meant to get removed in the meantime. Or if this is about resources that do get removed, then `0L` is a lie, the metric should stop being reported when the resource is actually gone.